Modules/_testclinic.c

```diff
@@ -0,0 +1,953 @@
+}
+
+
+/*[clinic input]
+size_t_converter
+
+    a: size_t = 12
+    /
+
+[clinic start generated code]*/
+
+static PyObject *
+size_t_converter_impl(PyObject *module, size_t a)
+/*[clinic end generated code: output=412b5b7334ab444d input=83ae7d9171fbf208]*/
+{
+    RETURN_PACKED_ARGS(1, PyLong_FromSize_t, size_t, a);
+}
+
+
+/*[clinic input]
+float_converter
+
+    a: float = 12.5
+    /
+
+[clinic start generated code]*/
+
+static PyObject *
+float_converter_impl(PyObject *module, float a)
+/*[clinic end generated code: output=1c98f64f2cf1d55c input=a625b59ad68047d8]*/
+{
+    RETURN_PACKED_ARGS(1, PyFloat_FromDouble, double, a);
+}
+
+
+/*[clinic input]
+double_converter
+
+    a: double = 12.5
+    /
+
+[clinic start generated code]*/
+
+static PyObject *
+double_converter_impl(PyObject *module, double a)
+/*[clinic end generated code: output=a4e8532d284d035d input=098df188f24e7c62]*/
+{
+    RETURN_PACKED_ARGS(1, PyFloat_FromDouble, double, a);
+}
+
+
+/*[clinic input]
+py_complex_converter
+
+    a: Py_complex
+    /
+
+[clinic start generated code]*/
+
+static PyObject *
+py_complex_converter_impl(PyObject *module, Py_complex a)
+/*[clinic end generated code: output=9e6ca2eb53b14846 input=e9148a8ca1dbf195]*/
+{
+    RETURN_PACKED_ARGS(1, PyComplex_FromCComplex, Py_complex, a);
+}
+
+
+/*[clinic input]
+str_converter
+
+    a: str = "a"
+    b: str(accept={robuffer}) = "b"
+    c: str(accept={robuffer, str}, zeroes=True) = "c"
+    /
+
+[clinic start generated code]*/
+
+static PyObject *
+str_converter_impl(PyObject *module, const char *a, const char *b,
+                   const char *c, Py_ssize_t c_length)
+/*[clinic end generated code: output=475bea40548c8cd6 input=bff2656c92ee25de]*/
+{
+    assert(!PyErr_Occurred());
+    PyObject *out[3] = {NULL,};
+    int i = 0;
+    PyObject *arg;
+
+    arg = PyUnicode_FromString(a);
+    assert(arg || PyErr_Occurred());
+    if (!arg) {
+        goto error;
+    }
+    out[i++] = arg;
+
+    arg = PyUnicode_FromString(b);
+    assert(arg || PyErr_Occurred());
+    if (!arg) {
+        goto error;
+    }
+    out[i++] = arg;
+
+    arg = PyUnicode_FromStringAndSize(c, c_length);
+    assert(arg || PyErr_Occurred());
+    if (!arg) {
+        goto error;
+    }
+    out[i++] = arg;
+
+    PyObject *tuple = PyTuple_New(3);
+    if (!tuple) {
+        goto error;
+    }
+    for (int j = 0; j < 3; j++) {
+        PyTuple_SET_ITEM(tuple, j, out[j]);
+    }
+    return tuple;
+
+error:
+    for (int j = 0; j < i; j++) {
+        Py_DECREF(out[j]);
+    }
+    return NULL;
+}
+
+
+static PyObject *
+bytes_from_buffer(Py_buffer *buf)
+{
+    PyObject *bytes_obj = PyBytes_FromStringAndSize(NULL, buf->len);
+    if (!bytes_obj) {
+        return NULL;
+    }
+    void *bytes_obj_buf = ((PyBytesObject *)bytes_obj)->ob_sval;
+    if (PyBuffer_ToContiguous(bytes_obj_buf, buf, buf->len, 'C') < 0) {
+        Py_DECREF(bytes_obj);
+        return NULL;
+    }
+    PyBuffer_Release(buf);
```

Comment:
This release should not be necessary; AC generates code to handle this in the wrapper.